### ISSUE TITLE
MKL: row_major in summary stats could be moved to general mkl enum for other oneMKL domains to use

### ISSUE THREAD

**Issue by spencerpatty** [oneapi-src/oneAPI-spec#326]
@paveldyakov I noticed that you have a `oneapi::mkl::stats::layout::row_major`  format defined.  At some point in sparse domain we are likely going to want to use the layout type variable as well.  Do you think it makes sense to move it to a general mkl enum clas type like `oneapi::mkl::layout::row_major` ?  (same for column_major).  

I understand that currently BLAS and LAPACK domains are currently using row_major and column_major namespaces so they might not need it, but who knows for the future if any type of dense matrix object is created... :)

**Comment by paveldyakov** [comment]
@spencerpatty, The proposal sounds great for me. I think it make sense in collaboration between stats and sparse domains on this question. 
Note: moving layout enum to the mkl namespace breaks backward compatibility for stats domain.

**Comment by spencerpatty** [comment]
@paveldyakov yeah it is true, I guess until the next major release we might have to go through process of adding an API that takes either mkl::layout  and deprecate the one that takes mkl::stats::layout but support both?

**Comment by paveldyakov** [comment]
@spencerpatty, yes, that is the right way here. We can add additional APIs for stats when mkl::layout is introduced. Spencer, am I right that 1.1 spec release is not a major release of specification?

**Comment by spencerpatty** [comment]
> am I right that 1.1 spec release is not a major release of specification?

Yes that is correct, this is a minor update, so deprecations are fine, and the timer starts for actual removal in a major release in a couple years....
